examples/bq-email-exports/main.py

```diff
@@ -49,87 +51,54 @@ def credentials():
     return credentials
 
 
-def main():
-    """Sends an email with a Google Cloud Storage signed URL of BQ Query results.
-    Creates BQ table, runs a SQL query, and exports the results to Cloud Storage as a CSV.
-    Generates signing credentials for the CSV and sends an email with a link to the signed URL.
-
-    Args:
-        None
-
-    Returns:
-        None
-    """
+def main(event,context):
+    """Entrypoint for Cloud Function"""
 
     # Create BQ and Storage Client
     bq_client = bigquery.Client(credentials=credentials())
     storage_client = storage.Client(credentials=credentials())
 
-    # Set variables
-    timestr = time.strftime("%Y%m%d%I%M%S")
-    project = "report-scheduling"
-    dataset_id = "bq_exports"
-    file_name = "daily_export_" + timestr
-    csv_name = file_name + ".csv"
-    table_id = "report-scheduling.bq_exports." + file_name
-    bucket_name = "bq_email_exports"
-    from_email = "sender@example.com"
-    to_emails = "recipient@example.com"
-
-    # Create a BQ table
-    schema = [
-        bigquery.SchemaField("url", "STRING", mode="REQUIRED"),
-        bigquery.SchemaField("view_count", "INTEGER", mode="REQUIRED"),
-    ]
-
-    # Make an API request to create table
-    table = bq_client.create_table(bigquery.Table(table_id, schema=schema))
-    print("Created table {}.{}.{}".format(table.project, table.dataset_id,
-                                          table.table_id))
-
-    # Run query on that table
-    job_config = bigquery.QueryJobConfig(destination=table_id)
+    # Get configurations from Cloud Scheduler payload
+    config = json.loads(base64.b64decode(event['data']).decode('utf-8'))
 
-    # Define the SQL query
-    sql = """
-        SELECT
-        CONCAT(
-            'https://stackoverflow.com/questions/',
-            CAST(id as STRING)) as url,
-        view_count
-        FROM `bigquery-public-data.stackoverflow.posts_questions`
-        WHERE tags like '%google-bigquery%'
-        ORDER BY view_count DESC
-        LIMIT 10
-    """
+    # Append timestamp to table name and set table_id 
+    table_name = config["table_name"] + time.strftime("%Y%m%d%I%M%S")
+    table_id = f"{config['project_id']}.{config['dataset_id']}.{table_name}"
+
+    # Get query 
+    job_config = bigquery.QueryJobConfig(destination=table_id)
```

Comment:
this should set [priority](https://googleapis.dev/python/bigquery/latest/generated/google.cloud.bigquery.job.QueryJobConfig.html#google.cloud.bigquery.job.QueryJobConfig.priority) to Batch as this is not an interactive use case.

Reply:
This is interactive use case since it's running within a cloud function and is blocking/waiting on results of the query.

Reply:
IIUC the use case itself is not interactive.
the chosen design of single blocking CF might make this seem interactive, but this means the design does not meet the use case.
See https://github.com/GoogleCloudPlatform/professional-services/pull/474#discussion_r426979469